domains/authorization/Authorization.API/API/Authorization/[Features]/GetUserOrganizationConsentsReceivedQuery.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Repository;
+using API.ValueObjects;
+using EnergyOrigin.Domain.ValueObjects;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Authorization._Features_;
+
+public class GetUserOrganizationConsentsReceivedQueryHandler(IOrganizationConsentRepository organizationConsentRepository)
+    : IRequestHandler<GetUserOrganizationConsentsReceivedQuery, GetUserOrganizationConsentsReceivedQueryResult>
+{
+    public async Task<GetUserOrganizationConsentsReceivedQueryResult> Handle(GetUserOrganizationConsentsReceivedQuery request, CancellationToken cancellationToken)
+    {
+        var userIdpUserIdClaim = IdpUserId.Create(Guid.Parse(request.IdpUserId));
```

Comment:
request.IdpUserId <-- Shouldn't we change this to be a guid, so we don't have to parse it in our request?

Reply:
Done